mimic/rest/maas_api.py

```diff
@@ -518,9 +518,21 @@ def delete_entity(self, request, tenant_id, entity_id):
         Delete an entity, all checks that belong to entity, all alarms that belong to those checks
         """
         entities = self._entity_cache_for_tenant(tenant_id).entities_list
+        previous_size = len(entities)
         entities[:] = [entity for entity in entities
                        if entity.id != entity_id]
 
+        if len(entities) == previous_size:
```

Comment:
@ksheedlo : I am aware that the length of the list changes when an item is deleted. (even without having to use intuition :) )

Has accessing an element when modifying the list or vice versa been an issue so far. I think we are over thinking this. All the plugins use the approach of finding an element from the list and we have even run performance tests and not had an issue.